### PR TITLE
Hotfix/1.1.1

### PR DESCRIPTION
**Pull request by Averus-a**
### Description of Change ###
Fix CL0002 and CL0006.
Fix code fix for CL0002.

### API Changes ###
 
None

### Platforms Affected ### 

- All

### Behavioral Changes ###

None

### PR Checklist ###

- [x] I have included examples or tests
- [ ] I have updated the change log
- [ ] I am listed in the CONTRIBUTORS file
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: src/CatenaLogic.Analyzers/Analyzers/CL0002/CL0002Analyzer.cs [hunk 1]
```diff
@@ -34,6 +36,12 @@ public override void HandleSyntaxNode(SyntaxNodeAnalysisContext context)
                 return;
             }
 
+            // Ignore Main method
+            if (string.Equals(methodSymbol.Name, DefaultMainMethodName) && methodSymbol.IsStatic)
```
Comment (GeertvanHorrik) [inline src/CatenaLogic.Analyzers/Analyzers/CL0002/CL0002Analyzer.cs:40]: for performance reasons, I recommend checking IsStatic first. Remember this runs on every compilation and every thing you type in, so we need to make it as fast as possible.

File: src/CatenaLogic.Analyzers/Analyzers/CL0002/CL0002CodeFixProvider.cs [hunk 2]
```diff
@@ -52,29 +53,37 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
 
             var root = await document.GetSyntaxRootAsync();
             if (root is null)
             {
-                return document;
+                return solution;
             }
 
-            var methodIdentifier = declarationSyntax.ChildTokens().FirstOrDefault(token => token.Kind() == SyntaxKind.IdentifierToken);
+            var methodSymbol = semanticModel.GetDeclaredSymbol(declarationSyntax, cancellationToken);
```
Comment (GeertvanHorrik) [inline src/CatenaLogic.Analyzers/Analyzers/CL0002/CL0002CodeFixProvider.cs:73]: remove empty line

File: src/CatenaLogic.Analyzers/Analyzers/CL0002/CL0002CodeFixProvider.cs [hunk 3]
```diff
@@ -52,29 +53,37 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
+            }
 
-            var newRoot = root.ReplaceToken(methodIdentifier, fixedMethodIdentifier);
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return solution;
+            }
 
-            var newDocument = document.WithSyntaxRoot(newRoot);
+            var modifiedSolution = await Renamer.RenameSymbolAsync(solution, methodSymbol, methodSymbol.Name + "Async", solution.Workspace.Options, cancellationToken);
```
Comment (GeertvanHorrik) [inline src/CatenaLogic.Analyzers/Analyzers/CL0002/CL0002CodeFixProvider.cs:85]: remove mepty line

File: src/CatenaLogic.Analyzers/Analyzers/CL0006/CL0006Analyzer.cs [hunk 4]
```diff
@@ -41,12 +41,62 @@ public override void HandleSyntaxNode(SyntaxNodeAnalysisContext context)
                     continue;
                 }
 
+                //note: Check is it part of simple lambda for an Expression parameter
+                var lambdaNode = FindParentExpresson<SimpleLambdaExpressionSyntax>(binaryExpressionSyntax, 7);
+
```
Comment (GeertvanHorrik) [inline src/CatenaLogic.Analyzers/Analyzers/CL0006/CL0006Analyzer.cs:46]: remove empty line

File: src/CatenaLogic.Analyzers/Analyzers/CL0006/CL0006Analyzer.cs [hunk 5]
```diff
@@ -41,12 +41,62 @@ public override void HandleSyntaxNode(SyntaxNodeAnalysisContext context)
...
+                if (lambdaNode is not null)
+                {
+                    var argumentTypeInfo = context.SemanticModel.GetTypeInfo(lambdaNode);
+
```
Comment (GeertvanHorrik) [inline src/CatenaLogic.Analyzers/Analyzers/CL0006/CL0006Analyzer.cs:50]: remove empty line

File: src/CatenaLogic.Analyzers/Analyzers/CL0006/CL0006Analyzer.cs [hunk 6]
```diff
@@ -41,12 +41,62 @@ public override void HandleSyntaxNode(SyntaxNodeAnalysisContext context)
...
+                        string.Equals(argumentTypeInfo.ConvertedType.ContainingNamespace.ToString(), "System.Linq.Expressions"))
+                    {
+                        return;
+                    }
+                }
+                else
+                {
+                    // Ignore inside query 
+                    var queryBody = binaryExpressionSyntax.Ancestors(false).FirstOrDefault(q => q is QueryBodySyntax) as QueryBodySyntax;
+                    if (queryBody != null)
```
Comment (GeertvanHorrik) [inline src/CatenaLogic.Analyzers/Analyzers/CL0006/CL0006Analyzer.cs:66]: is not null :-)

File: src/CatenaLogic.Analyzers/Analyzers/CL0006/CL0006Analyzer.cs [hunk 7]
```diff
@@ -41,12 +41,62 @@ public override void HandleSyntaxNode(SyntaxNodeAnalysisContext context)
...
+                return null;
+            }
+
+            var parent = node.Parent;
+            if (parent is null)
+            {
+                return null;
+            }
+
+            if (parent is T)
```
Comment (GeertvanHorrik) [inline src/CatenaLogic.Analyzers/Analyzers/CL0006/CL0006Analyzer.cs:94]: if (parent is T typedParent)
{
return typedParent;
}

**Review by GeertvanHorrik — CHANGES_REQUESTED** [review]

File: src/CatenaLogic.Analyzers/Analyzers/CL0002/CL0002Analyzer.cs (hunk 1 above)
Comment (Averus-a) [reply src/CatenaLogic.Analyzers/Analyzers/CL0002/CL0002Analyzer.cs:40]: Great note

**Review by GeertvanHorrik — APPROVED** [review]

**Pull request closed without merging**